cmd/sink/main.go

```diff
@@ -61,89 +57,107 @@ func NewSink(db database.DBInterface, logger *log.Logger, filters *filters.Filte
 		}),
 	)
 	if err != nil {
-		logger.Fatalf("Failed to create HTTP client: %s\n", err.Error())
+		slog.Error("Failed to create HTTP client", "err", err.Error())
+		os.Exit(1)
 	}
 	eventClient, err := cloudevents.NewClient(httpClient, ceClient.WithObservabilityService(ceOtelObs.NewOTelObservabilityService()))
 	if err != nil {
-		logger.Fatalf("Failed to create CloudEvents HTTP client: %s\n", err.Error())
+		slog.Error("Failed to create CloudEvents HTTP client", "err", err.Error())
+		os.Exit(1)
 	}
 
 	k8sClient, err := k8s.GetKubernetesClient()
 	if err != nil {
-		logger.Fatalln("Could not start a kubernetes client:", err)
+		slog.Error("Could not start a kubernetes client", "err", err)
+		os.Exit(1)
 	}
 
 	return &Sink{
 		Db:          db,
 		EventClient: eventClient,
 		Filters:     filters,
 		K8sClient:   k8sClient,
-		Logger:      logger,
 	}
 }
 
 // Processes incoming cloudevents and writes them to the database
 func (sink *Sink) Receive(ctx context.Context, event cloudevents.Event) {
-	sink.Logger.Println("Received cloudevent: ", event.ID())
+	slog.Info("Received cloudevent", "id", event.ID())
 	k8sObj, err := models.UnstructuredFromByteSlice(event.Data())
 	if err != nil {
-		sink.Logger.Printf("Cloudevent %s is malformed and will not be processed: %s\n", event.ID(), err)
+		slog.Error("Cloudevent is malformed and will not be processed", "id", event.ID(), "err", err)
```

Comment:
Errors indicate a failure - since you are not failing at this point, shouldn't this be a warning (`slog.Warn`) instead of an error?

Reply:
A warning could be understood as something we don't need to address, I don't know if here that would be true. If running we get this problem a lot and its just life, we can use Warn instead.